src/lib/duration/format.js

```diff
@@ -0,0 +1,283 @@
+                switch (decVal.length) {
+                    case 1:
+                        decimalValue = (decVal[0] + zeroFill(0, settings.precision)).slice(0, settings.precision);
+                        break;
+
+                    case 2:
+                        decimalValue = (decVal[1] + zeroFill(0, settings.precision)).slice(0, settings.precision);
+                        break;
+
+                    case 3:
+                        decimalValue = (zeroFill(0, (+decVal[2]) - 1) + (decVal[0] || '0') + decVal[1] + zeroFill(0, settings.precision)).slice(0, settings.precision);
+                        break;
+
+                    default:
+                        break;
+                    //     throw 'Moment Duration Format: unable to parse token decimal value.';
+                    // TODO: what should we do instead of throwing an error?
+                }
+            }
+        } else {
+            wholeValue = Math[truncMethod](value);
+            decimalValue = value - wholeValue;
+        }
+
+        // update tokens array
+        // using this algorithm to not assume anything about the order or frequency of any tokens
+        each.call(tokens, function (token) {
+            if (token.type === momentType) {
+                extend(token, {
+                    value: value,
+                    wholeValue: wholeValue,
+                    decimalValue: decimalValue,
+                    isLeast: isLeast,
+                    isMost: isMost
+                });
+            }
+        });
+
+        // update remainder
+        remainder.subtract(wholeValue, momentType);
+    });
+
+    // build output
+
+    // the first moment token can have special handling
+    foundFirst = false;
+
+    tokens = map(tokens, function (token, index) {
+        var val,
+            decVal;
+
+        if (!token.type) {
+            // if it is not a moment token, use the token as its own value
+            return token.token;
+        }
+
+        // remove negative sign from the beginning
+        val = token.wholeValue.toString().replace(/^\-/, '');
+
+        // apply token length formatting
+        // special handling for the first moment token that is not the most significant in a trimmed template
+        if (token.length > 1 && (foundFirst || token.isMost)) {
+            val = zeroFill(val, token.length);
+        }
+
+        // add decimal value if precision > 0
+        if (token.isLeast && (settings.precision > 0)) {
+            decVal = token.decimalValue.toString();
+            val += settings.decimalSeparator + decVal;
+        }
+
+        // add a negative sign if the value is negative and token is the first one
+        if (token.value < 0 && !index) {
+            val = '-' + val;
+        }
+
+        foundFirst = true;
+
+        return val;
+    });
+
+
+    return this.localeData().postformat(tokens.join(''));
+}
+
+formatDuration.defaults = {
+    // token definitions
```

Comment:
If this is coming into core, it needs to use the same format tokens that Moment does or we will be crazy confused from a support standpoint. If you change it to do that, it can use the same regular expressions, imported from the units folder, to look for the tokens